test/e2e/specs/interactivity/router-styles.spec.ts

```diff
@@ -285,4 +285,29 @@ test.describe( 'Router styles', () => {
 		await expect( red ).toHaveCSS( 'color', COLOR_RED );
 		await expect( redBlock ).toBeVisible();
 	} );
+
+	test( 'should not apply preloaded styles in current page', async ( {
+		page,
+	} ) => {
+		const red = page.getByTestId( 'red-from-inline' );
+		const green = page.getByTestId( 'green-from-inline' );
+		const blue = page.getByTestId( 'blue-from-inline' );
+		const all = page.getByTestId( 'all-from-inline' );
+		const prefetching = page.getByTestId( 'prefetching' );
+
+		await expect( red ).toHaveCSS( 'color', COLOR_WRAPPER );
+		await expect( green ).toHaveCSS( 'color', COLOR_WRAPPER );
+		await expect( blue ).toHaveCSS( 'color', COLOR_WRAPPER );
+		await expect( all ).toHaveCSS( 'color', COLOR_WRAPPER );
+
+		await page.getByTestId( 'link red' ).hover();
+
+		// Wait until the prefetching has finished.
+		await expect( prefetching ).toHaveText( 'false' );
```

Comment:
We should probably check if `prefetching` starts with the value `true` to make sure that it's changing.

Reply:
Absolutely! Fixed in https://github.com/woocommerce/gutenberg/pull/9/commits/3fd9ba7d6152faaaa1fcd1f6ee9b3ba0d758c887.